cmd/update/volume_move_replica.go

```diff
@@ -122,33 +124,89 @@ func (c *volumeMoveReplicaCommand) runWithCtx(ctx context.Context, cmd *cobra.Co
 		}
 	}
 
-	// wait for desired state (new replica on target node and is synced)
-	err = c.waitForNewReplica(ctx, nsID)
+	// wait for a new replica on [destination] node
+	err = c.waitForReplicaOnNode(ctx, nsID)
 	if err != nil {
 		return err
 	}
 
+	// get updated volume config
+	vol, err = c.getVolume(ctx, nsID)
+	if err != nil {
+		return fmt.Errorf("failed to get updated volume config from api: %s", err)
+	}
+
 	// mark the deployments on the source node as preferred for eviction
 	params := &apiclient.SetPreferredEvictionCandidatesRequestParams{
 		DeploymentIDs: deploymentsOnSourceNode,
 	}
 	if c.useCAS() {
-		params.CASVersion = version.FromString(c.casVersion)
+		params.CASVersion = vol.Version
 	}
 	err = c.client.SetPreferredEvictionCandidates(ctx, nsID.String(), vol.ID.String(), params)
 	if err != nil {
 		return fmt.Errorf("failed to set preferred eviction candidates: %s", err)
 	}
 
-	c.display.TextMessage(ctx, c.writer, "deployments on [from] node marked as preferred for eviction")
+	c.display.TextMessage(ctx, c.writer, "deployments on [source] node marked as preferred for eviction")
+
+	// wait for the new replica on [destination] node to sync
+	err = c.waitForReplicaOnNodeToSync(ctx, nsID)
+	if err != nil {
+		return err
+	}
+
+	c.display.TextMessage(ctx, c.writer, "replica on [destination] node finished syncing")
+
+	if !isMovingMaster {
+		return nil
+	}
+
+	// find the deployment ID of the newly created one from the call above
+	var newReplicaID id.Deployment
+	for _, rep := range vol.Replicas {
+		if rep.Node.String() == c.toNode {
+			newReplicaID = rep.ID
+		}
+	}
+
+	err = c.client.AttemptPromotion(ctx, string(nsID), vol.ID.String(), newReplicaID.String())
+	if err != nil {
+		return err
+	}
+
```

Comment:
I think we should do the promotion as the first step. Promoting a random replica to master. 

If we do it this way we can once again be wrecked by the api-manager trying to scale down the deployments. 

Also, `SetPreferredEvictionCandidates` will fail if we attempt to set the master as a preferred eviction candidate